ome/resource_deploy.go

```diff
@@ -10,195 +10,195 @@ import (
+	resp.TypeName = req.ProviderTypeName + "deployment"
+}
 
 // Template Deployment Resource schema
-func (r resourceDeploymentType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
-	return tfsdk.Schema{
+func (r resourceDeployment) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
 		MarkdownDescription: "Resource for managing template deployment on OpenManage Enterprise. Updates are supported for the following parameters: `device_ids`, `device_servicetags`, `boot_to_network_iso`, `forced_shutdown`, `options_time_to_wait_before_shutdown`, `power_state_off`, `options_precheck_only`, `options_strict_checking_vlan`, `options_continue_on_warning`, `run_later`, `cron`, `device_attributes`, `job_retry_count`, `sleep_interval`.",
 		Version:             1,
-		Attributes: map[string]tfsdk.Attribute{
-			"id": {
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
 				MarkdownDescription: "ID of the resource.",
 				Description:         "ID of the resource.",
-				Type:                types.StringType,
 				Computed:            true,
 			},
-			"template_id": {
+			"template_id": schema.Int64Attribute{
 				MarkdownDescription: "ID of the existing template.",
 				Description:         "ID of the existing template.",
-				Type:                types.Int64Type,
 				Computed:            true,
 				Optional:            true,
-				PlanModifiers: tfsdk.AttributePlanModifiers{
-					tfsdk.UseStateForUnknown(),
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
 				},
 			},
-			"template_name": {
+			"template_name": schema.StringAttribute{
 				MarkdownDescription: "Name of the existing template.",
 				Description:         "Name of the existing template.",
-				Type:                types.StringType,
 				Optional:            true,
 				Computed:            true,
-				PlanModifiers: tfsdk.AttributePlanModifiers{
-					tfsdk.UseStateForUnknown(),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
-			"device_ids": {
+			"device_ids": schema.SetAttribute{
 				MarkdownDescription: "List of the device id(s).",
 				Description:         "List of the device id(s).",
-				Type: types.SetType{
-					ElemType: types.Int64Type,
-				},
-				Optional: true,
+				ElementType:         types.Int64Type,
+				Optional:            true,
 			},
-			"device_servicetags": {
+			"device_servicetags": schema.SetAttribute{
 				MarkdownDescription: "List of the device servicetags.",
 				Description:         "List of the device servicetags.",
-				Type: types.SetType{
-					ElemType: types.StringType,
-				},
-				Optional: true,
+				ElementType:         types.StringType,
+				Optional:            true,
 			},
-			"boot_to_network_iso": {
+			"boot_to_network_iso": schema.ObjectAttribute{
```

Comment:
Its recommended in https://developer.hashicorp.com/terraform/plugin/framework/handling-data/attributes#object to use single nested attribute (and we have done so before as well)
Ref: https://developer.hashicorp.com/terraform/plugin/framework/handling-data/attributes#singlenestedattribute

Reply:
If this requires too much refactoring, then that can be taken up in a separate effort.

Reply:
I have checked both implementations,
it looks single nested attribute offers a few more methods.
But deciding factor will be, whether are we going to use those functions or not.
https://github.com/hashicorp/terraform-plugin-framework/blob/8536f49ae31ef761e3ab0a74e69fd31f4ec0fd6a/provider/metaschema/object_attribute.go
https://github.com/hashicorp/terraform-plugin-framework/blob/8536f49ae31ef761e3ab0a74e69fd31f4ec0fd6a/provider/metaschema/single_nested_attribute.go
![image](https://user-images.githubusercontent.com/118425422/223383351-a3b65384-5175-4b4f-8363-aacea52876ba.png)

Reply:
we can take require this kind of refactoring to all the instances
![image](https://user-images.githubusercontent.com/118425422/223388299-951e5aa7-b137-4146-abec-91ee8ff7948a.png)

Reply:
we may take this refactoring in the next PR.